### DISCUSSION TITLE
Action–Domain–Responder pattern in the future.

### DISCUSSION THREAD

**Discussion by AliShareei** [phalcon/phalcon#91] [General] [answered]
Hi guys.

Mr @niden in #86 talk about switch from MVC to ADR pattern.
I know this question may be too early but The question for me was, will Phalcon lose the MVC feature or will we have ADR along with the MVC feature together?

**Comment by niden** [comment, marked as answer]
@AliShareei For one major version both patterns will be available to allow more time for developers to adjust their applications. After that, _most likely_ we will drop MVC. 

The timeframe for this is after v5 so plenty of time ahead of us. The reason I wrote _most likely_ is because things change and we will change with them. As things are that is the plan but time will also tell if we need to change that approach.
